Use conda-build's conda interface

jakirkham opened the pull request:
Grab a few things we were grabbing from `conda` from `conda_build.conda_interface` instead. This should smooth over any remaining issues between `conda` 4.3 and 4.4+.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

jakirkham committed:
Use conda-build 3 to get `subdir`

In `conda` 4.4+, `conda.config` has been dropped. To address this issue,
use `conda_build`'s `conda_interface` module to smooth over this
difference between `conda` 4.3 and 4.4.

jakirkham committed:
Use `get_index` from `conda_build.conda_interface`

To patch over some other differences between `conda` 4.3 and 4.4, import
`get_index` from `conda_build.conda_interface` as it's location varies
between `conda` 4.3 and 4.4+.

jakirkham committed:
Bump to 4.4.22

Instead of using `conda` for a few API functions, just pull them from
`conda_build.conda_interface` to standardize over `conda` versions.
Rebuild and release so that feedstocks can benefit from this
compatibility fix.

jakirkham commented:
If you have a second, could you please give this a quick look @msarahan? This would help current feedstocks.

jakirkham commented:
Thanks Mike. :)
